### ISSUE TITLE
Runtime warnings after #855

### ISSUE THREAD

**Issue by hjmjohnson** [InsightSoftwareConsortium/ITK#938]
### Description
Related to #855 

New runtime warnings are produced, but results appear to be OK.

```
WARNING: In /Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx, line 88
LabelStatisticsImageFilter (0x278c2d0): Unable to convert input "LabelInput" to type N3itk5ImageIdLj3EEE

WARNING: In /Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx, line 88
ImageToHistogramFilter (0x279bd70): Unable to convert input "AutoMinimumMaximum" to type N3itk5ImageIsLj3EEE

WARNING: In /Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx, line 88
ImageToHistogramFilter (0x279bd70): Unable to convert input "HistogramSize" to type N3itk5ImageIsLj3EEE

WARNING: In /Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx, line 88
ImageToHistogramFilter (0x279bd70): Unable to convert input "MarginalScale" to type N3itk5ImageIsLj3EEE
```

### Steps to Reproduce

Run BRAINSConstellationDetector software on any data set.

### Expected behavior

No runtime warnings are produced.

### Actual behavior

Runtime warnings are produced.

### Reproducibility

100%

### Versions
Versoins after #855 

### Environment
Linux release build

### Additional Information

If this a real warning, it would be nice to have a recommendation for how to resolve.

**Comment by blowekamp** [comment]
@hjmjohnson Could you give this a try:
```
diff --git a/Modules/Core/Common/include/itkImageSink.hxx b/Modules/Core/Common/include/itkImageSink.hxx
index 1e11220a36..a47755bac3 100644
--- a/Modules/Core/Common/include/itkImageSink.hxx
+++ b/Modules/Core/Common/include/itkImageSink.hxx
@@ -158,7 +158,7 @@ ImageSink<TInputImage>
       // static_casts the input to an TInputImage).
       using ImageBaseType = ImageBase< InputImageDimension >;
       typename ImageBaseType::ConstPointer constInput =
-        dynamic_cast< ImageBaseType const * >( this->GetInput(inputName) );
+        dynamic_cast< ImageBaseType const * >( this->ProcessObject::GetInput(inputName) );
 
       // If not an image, skip it, and let a subclass of
       // ImageToImageFilter handle this input.
@@ -169,7 +169,7 @@ ImageSink<TInputImage>
 
       // Input is an image, cast away the constness so we can set
       // the requested region.
-      InputImagePointer input = const_cast< TInputImage * >( this->GetInput(inputName) );
+      InputImagePointer input = const_cast< TInputImage * >( this->ProcessObject::GetInput(inputName) );
 
       // copy the requested region of the first input to the others
       InputImageRegionType inputRegion;
```

**Comment by hjmjohnson** [comment]
No love :(

```
                 from /Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/BRAINSTools/BRAINSCommonLib/BRAINSFitHelper.cxx:20:
/Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx: In instantiation of 'void itk::ImageSink<TInputImage>::GenerateNthInputRequestedRegion(unsigned int) [with TInputImage = itk::Image<double, 3u>]':
/Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/BRAINSTools/BRAINSCommonLib/BRAINSFitHelper.cxx:798:1:   required from here
/Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx:172:103: error: invalid const_cast from type 'itk::DataObject*' to type 'itk::Image<double, 3u>*'
       InputImagePointer input = const_cast< TInputImage * >( this->ProcessObject::GetInput(inputName) );
                                                                                                       ^
/Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx: In instantiation of 'void itk::ImageSink<TInputImage>::GenerateNthInputRequestedRegion(unsigned int) [with TInputImage = itk::Image<float, 3u>]':
/Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/BRAINSTools/BRAINSCommonLib/BRAINSFitHelper.cxx:798:1:   required from here
/Shared/johnsonhj/Binaries/20190513/Linux/NEP-14/ITKv5/Modules/Core/Common/include/itkImageSink.hxx:172:103: error: invalid const_cast from type 'itk::DataObject*' to type 'itk::Image<float, 3u>*'
make[2]: *** [BRAINSCommonLib/CMakeFiles/BRAINSCommonLib.dir/BRAINSFitHelper.cxx.o] Error 1
make[1]: *** [BRAINSCommonLib/CMakeFiles/BRAINSCommonLib.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by blowekamp** [comment]
If there is a quicker way to reproduce this problem, then it can be addressed more quickly.

**Comment by blowekamp** [comment]
Some of these warning occur with the `itkImageToHistogramFilterTest`.

**Comment by romangrothausmann** [comment]
In my CLI tests (which are far from exaustive) the warnings only appear for `itkImageToHistogramFilter` (https://gitlab.com/romangrothausmann/ITK-CLIs/-/jobs/217332000), not e.g. for `itkStatisticsImageFilter` (https://gitlab.com/romangrothausmann/ITK-CLIs/-/jobs/217331998)
They appear for  `itkImageToHistogramFilter` with and without streaming (https://gitlab.com/romangrothausmann/ITK-CLIs/-/jobs/217331999), without streaming only at the start of the execution, while with streaming apparently for every chunk.
